swig_obj: unreferenced local variable

Generation of code using `-python -builtin` flags on Swig 4.0.1 makes Visual Studio 2017 raise a bunch of warnings:
```
warning C4101: 'swig_obj': unreferenced local variable
```
This seems to happen for functions that wrap a member method without parameters, like delete, get, cast, ...

--- a/Source/Modules/python.cxx
+++ b/Source/Modules/python.cxx
@@ -2810,7 +2810,7 @@ public:
 
     /* Generate code for argument marshalling */
     if (funpack) {
-      if (num_arguments > 0 && !overname) {
+      if (num_arguments > (builtin_self && !constructor ? 1 : 0) && !overname) {
 	sprintf(source, "PyObject *swig_obj[%d]", num_arguments);
 	Wrapper_add_localv(f, "swig_obj", source, NIL);
       }

